### PR TITLE
Carry comment over if test suite failed with same serial failure

### PR DESCRIPTION
**Pull request by rwx788**
Follows [PR#998](https://github.com/os-autoinst/os-autoinst/pull/998) in os-autoinst (not yet merged).
Our use case is that we have kernel bug, which fails different modules,
depending on our luck. Therefore, automatic comment carryover won't help
to the reviewers and they have to do monkey job to label them.

With this change, if failure reason will be marked with bug reference,
it will be treated as a reason of test suite failure. Therefore, if same
failure is thrown in multiple runs, irregardless the module where it was
detected, last comment will be taken over.

See [poo#36448](https://progress.opensuse.org/issues/36448).

[Verification run](http://g226.suse.de/tests/2176#next_previous). Here you can see that test failed in 2 different modules, but comment from the first one was added to the last.

This PR doesn't yet include documentation update and a test.

### CODE REVIEW

**Comment by coolo** [comment]
I approve - but we need a test

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -1559,17 +1559,26 @@ sub _failure_reason {
 
     my @failed_modules;
     my $modules = $self->modules;
+
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Unrelated whitespace change I would avoid.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 2]
```diff
@@ -1559,17 +1559,26 @@ sub _failure_reason {
...
     while (my $m = $modules->next) {
         if ($m->result eq FAILED || $m->result eq SOFTFAILED) {
-            push(@failed_modules, $m->name . ':' . $m->result);
+            # Look for serial failures which have bug reference
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: That would not really be limited to "serial failures" so I guess you can get rid of this comment

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 3]
```diff
@@ -1559,17 +1559,26 @@ sub _failure_reason {
...
+            my @bugrefs = map { find_bugref($_->{title}) || '' } @{$m->details};
+            # If bug reference is in title, put it as a failure reason, otherwise use module name
+            if (my $failure_reason = join('', @bugrefs)) {
+                return $failure_reason;
+            }
+            else {
+                push(@failed_modules, $m->name . ':' . $m->result);
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Hm, can you think of a case where we would find a bugref above but actually want this else-branch to be active? E.g. Can we still override a failed job with a a bugref that a human reviewer specifies for carryover?

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (rwx788) [reply lib/OpenQA/Schema/Result/Jobs.pm]: I don't like when different logical parts having code on each line, so I would rather keep it. It's a matter of taste.

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 2 above)
Comment (rwx788) [reply lib/OpenQA/Schema/Result/Jobs.pm]: It would, because as of now `record_resultfile` with custom message is called only in case of serial failure.

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 3 above)
Comment (rwx788) [reply lib/OpenQA/Schema/Result/Jobs.pm]: Yes, that's the whole point for this approach, as it's just failure detection, but we don't change/add the comment, it still has to be given by the reviewer. I don't see any scenarios except these use-cases with kernel bugs where this feature is useful, but we already have 2 of them (one for 42.3 leap upgrade test and second one for sle on arm).

**Comment by foursixnine** [comment]
Since this one is merged PR#998 needs to be merged too
